modules/data-streams/src/yamlRestTest/java/org/elasticsearch/datastreams/DataStreamsClientYamlTestSuiteIT.java

```diff
@@ -48,8 +60,12 @@ private static ElasticsearchCluster createCluster() {
             .distribution(DistributionType.DEFAULT)
             .feature(FAILURE_STORE_ENABLED)
             .setting("xpack.security.enabled", "true")
-            .keystore("bootstrap.password", "x-pack-test-password")
-            .user("x_pack_rest_user", "x-pack-test-password");
+            .keystore("bootstrap.password", PASS)
+            .user("x_pack_rest_user", PASS)
+            .user("data_stream_test_user", PASS, "data_stream_alias_test_role", true)
```

Comment:
The same data stream YAML tests are running in Serveless. Marked `data_stream_test_user` as operator to allow setting `index.number_of_replicas` index setting.